Code/Framework/AzCore/AzCore/Serialization/ObjectStream.cpp

```diff
@@ -281,7 +281,16 @@ namespace AZ
 
                 if (childClass)
                 {
-                    AZ_Assert(childNode.m_element.m_version <= childClass->m_version, "Serialize was parsing old version class and found newer version element! This should be impossible!");
+                    AZ_Error("Error", childNode.m_element.m_version <= childClass->m_version,
+                        "The class %s supports version %d. Found file class %s with version %d during deserializing, which is a newer version than the code supports. "
+                        "First, check if you've built latest, your C++ could be out of date. "
```

Comment:
```suggestion
                        "First, check if you've built latest, your C++ code could be out of date. "
```

Reply:
Unfortunately there the "how to fix it" comes in with many explanations how the data got into this state.